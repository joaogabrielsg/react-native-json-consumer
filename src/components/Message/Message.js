import React from 'react';
import PropTypes from 'prop-types';
import { Platform, StyleSheet, Text, View, Modal } from 'react-native';

import { metrics, colors, fonts } from '../../styles';
import Button from '../Button/Button';

const Message = ({ title, message, buttonText, open, onClose }) => (
  <View style={styles.container}>
    <Modal visible={open} transparent={true} animationType={'fade'}>
      <View style={styles.fullScreenView}>
        <View style={styles.messageView}>
          <View style={styles.messageContent}>
            <Text style={styles.title}>{title}</Text>
            <Text style={styles.messageText} numberOfLines={3}>
              {message}
            </Text>
            <Button color="primary" title={buttonText} onPress={onClose} />
          </View>
        </View>
      </View>
    </Modal>
  </View>
);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    position: 'absolute',
    justifyContent: 'center',
    alignItems: 'center',
    marginTop: metrics.statusBarHeight
  },
  fullScreenView: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: 'rgba(52, 52, 52, 0.6)'
  },
  messageView: {
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: colors.messageBackground,
    height: metrics.alertHeight,
    width: '90%',
    borderRadius: metrics.borderRadius
  },
  messageContent: {
    alignItems: 'center',
    justifyContent: 'space-between',
    height: '70%',
    width: '90%'
  },
  title: {
    color: colors.messageText,
    fontSize: fonts.messageTitle,
    fontWeight: '800'
  },
  messageText: {
    textAlign: 'center',
    color: colors.messageText,
    fontSize: fonts.messageText
  }
});

Message.propTypes = {
  title: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired,
  buttonText: PropTypes.string.isRequired,
  open: PropTypes.bool.isRequired
};

export default Message;
