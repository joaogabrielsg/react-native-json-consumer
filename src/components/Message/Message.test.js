import 'react-native';
import React from 'react';
import Message from './Message';

import renderer from 'react-test-renderer';

describe('Message', () => {
  it('sholud render correctly', () => {
    const message = renderer
      .create(<Message title="Test" message="test" buttonText="testButton" open={false} />)
      .toJSON();
    expect(message).toMatchSnapshot();
  });
});
