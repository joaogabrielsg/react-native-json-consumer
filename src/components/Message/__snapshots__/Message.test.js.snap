// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message sholud render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "marginTop": 20,
      "position": "absolute",
    }
  }
>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(52, 52, 52, 0.6)",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgb(32, 34, 52)",
            "borderRadius": 12,
            "height": 220,
            "justifyContent": "center",
            "width": "90%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "height": "70%",
              "justifyContent": "space-between",
              "width": "90%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "rgb(255, 255, 255)",
                "fontSize": 26,
                "fontWeight": "800",
              }
            }
          >
            Test
          </Text>
          <Text
            numberOfLines={3}
            style={
              Object {
                "color": "rgb(255, 255, 255)",
                "fontSize": 16,
                "textAlign": "center",
              }
            }
          >
            test
          </Text>
          <View
            style={
              Object {
                "width": 600,
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgb(0,	208, 229)",
                  "borderRadius": 12,
                  "borderWidth": 0,
                  "opacity": 1,
                  "padding": 33.35,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 17,
                    "fontWeight": "bold",
                  }
                }
              >
                testButton
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
